.github/workflows/CI.yml

```diff
@@ -69,18 +69,14 @@ jobs:
       - name: "GAP tests"
         run: |
           julia --project=. -e 'import GAP; GAP.create_gap_sh("/tmp")'
-          export GAP="/tmp/gap.sh -A --quitonbreak --norepl"
+          export GAP="/tmp/gap.sh --quitonbreak --norepl"
```

Comment:
So this is a nasty hack: because we specify `-A` now any places that already specifies `-A` needs to remove it... But IMHo that solution doesn't scale well.

Alternative ideas for resolving this:
1. we filter the args being passed in to remove any `-A` arguments
2. we don't use `-A` to reduce the loaded packages, but rather `--bare`
   - optionally we could afterwards "manually" load the required packages gapdoc/smallgrp/transgrp/primgrp
3. we don't use `-A` nor `--bare` and instead patch the GAP library directly to support the "overrides" (i.e. `GAP_lib_jll` for now and possibly later upstream GAP)
   - if we do it right then `DirectoriesPackageProgramsOverrides` exists by the time `gap/systemfile.g` is executed and we can fill it then. 
   - this would also allow us to load the default set of GAP packages again (I am torn whether we want that, or want `--bare`, or `-A` ...)
4. as a variant of that we could also use another dirty hack and run the code from `gap/pkg.g` (which patches `DirectoriesPackagePrograms`) from inside `pkg/JuliaInterface/PackageInfo.g` after making sure that `JuliaInterface` is the first package to be loaded...

Reply:
Currently my point of view is:

When starting GAP from GAP.jl then we should  load JuliaInterface but no other GAP packages.
This way we allow Oscar.jl (or others) to prepare special settings that will influence what happens when certain GAP packages will get loaded, and it is possible to provide/prescribe special packages versions and perhaps to load these packages.

We can provide a small function in GAP.jl that triggers GAP's `LoadPackages` for those packages that are "wanted by GAP" and that have not yet been loaded.
The question is then how this function can be called automatically but late enough such that the above order of package loading does not get distorted.